usd/src/usd_parser/USDWorld.cc

```diff
@@ -67,10 +76,66 @@ namespace usd
       _world.SetName(worldName + "_world");
     }
 
+    std::string linkName;
+
     auto range = pxr::UsdPrimRange::Stage(reference);
     for (auto const &prim : range)
     {
+      // Skip materials, the data is already available in the USDData class
+      if (prim.IsA<pxr::UsdShadeMaterial>() || prim.IsA<pxr::UsdShadeShader>())
+      {
+        continue;
+      }
+
       std::string primName = prim.GetName();
+      std::string primPath = pxr::TfStringify(prim.GetPath());
+      std::string primType = prim.GetPrimTypeInfo().GetTypeName().GetText();
+
+      std::vector<std::string> primPathTokens =
+        ignition::common::split(primPath, "/");
+
+      // In general USD models used in Issac Sim define the model path
+      // under a root path for example:
+      //  -> /robot_name/robot_name_link0
+      // But sometimes for enviroments it uses just a simple path:
+      //  -> /ground_plane
+      //  -> /wall_0
+      // the shortName variable defines if this is the first case when it's
+      // False or when it's true then it's the second case.
+      // This conversion might only work with Issac Sim USDs
+      if (primPathTokens.size() >= 2)
+      {
+        bool shortName = false;
+        if (primPathTokens.size() == 2)
+        {
+          if (prim.IsA<pxr::UsdGeomGprim>() || (primType == "Plane"))
+          {
+            if (primName != "imu")
+            {
+              linkName = "/" + primPathTokens[0];
+              shortName = true;
+            }
+          }
+        }
+        if (!shortName)
+        {
+          linkName = "/" + primPathTokens[0] + "/" + primPathTokens[1];
+        }
+      }
```

Comment:
Is this code mainly used for finding the parent prim of the light? I find it a little odd that we are manually parsing through the first 2 levels of the prim path to find the parent. I imagine this could be problematic if the light's parent isn't in the first two levels of the stage. Is there a better way to find the parent of a light that offers more flexibility/is less error prone?

Reply:
I didn't find a better way to do it, I'm open to suggestions

Reply:
I can't think of a better way at the moment - since we are pressed for time, I added a TODO note in https://github.com/ignitionrobotics/sdformat/pull/875/commits/847271a1821043f64cefab13c0c7219a43d6cb6e and created an issue in #927.